### PR TITLE
disallow submitting disabled/archived/expired catalogue items

### LINKED ISSUE #1524: Shouldn't be able to submit application if catalogue item is disabled/archived/expired
... but submitted applications can still be handled.

### PR DESCRIPTION
**Pull request by opqdonut**
fixes #1524

### CODE REVIEW

File: src/clj/rems/db/applications.clj [hunk 1]
```diff
@@ -100,6 +100,10 @@
 (mount/defstate application-cache
   :start (atom (cache/ttl-cache-factory {} :ttl 10000)))
 
+;; TODO combine with reload-cache!?
+(defn reset-application-cache! []
+  (swap! application-cache empty))
```
Comment (luontola) [inline src/clj/rems/db/applications.clj]: Yes, better combine with `reload-cache!`.

File: test/clj/rems/api/test_applications.clj [hunk 2]
```diff
@@ -432,6 +434,38 @@
...
+(deftest test-application-submit
+  (let [user-id "alice"
+        form-id (test-data/create-form! {})
+        cat-id (test-data/create-catalogue-item! {:form-id form-id})
+        app-id (test-data/create-application! {:catalogue-item-ids [cat-id] :actor user-id})]
+    (testing "submit with disabled catalogue item fails"
+      (is (:success (catalogue/update-catalogue-item! {:id cat-id
+                                                       :enabled false
+                                                       :archived false})))
+      (rems.db.applications/reset-application-cache!)
```
Comment (luontola) [inline test/clj/rems/api/test_applications.clj]: For this test you could stop rems.db.applications/application-cache, in which case the reads would happen without a cache.

Comment (opqdonut) [reply test/clj/rems/api/test_applications.clj]: I think having an explicit cache reset makes the problem more visible so I'll leave it.

File: src/clj/rems/db/applications.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/db/applications.clj]: I'll do this

**Pull request merged**